Sources/App/Controllers/BusController.swift

```diff
@@ -19,8 +19,9 @@ final class BusController: RouteCollection {
     func getBusData(req: Request) throws -> Future<DataResponse> {
         let transloc = try req.make(TranslocService.self)
         let stopId = try req.parameters.next(String.self)
+        let route  = try req.query.get(String.self, at: "route")
```

Comment:
Now that the function this gets passed into is optional (see my comments below) you should change this to:

```swift
let route  = try? req.query.get(String.self, at: "route")
```

The `try?` tells it to pass `nil` to the `route` constant instead of throwing an error if it doesn't find a `route` query parameter. This makes it backwards-compatible so that existing clients using the API don't have to change logic if they don't want to.